packages/design-system/src/components/Modal/Modal.tsx

```diff
@@ -30,22 +30,27 @@ export type ModalPropsType = {
 		icon?: ModalIcon;
 	};
 	onClose: Function;
+	disclosure?: Function;
 	primaryAction?: ButtonPrimaryPropsType | { destructive: true & ButtonDestructivePropsType };
 	secondaryAction?: ButtonSecondaryPropsType;
 	preventEscaping?: boolean;
 	children: ReactNode | ReactNode[];
 };
 
 function Modal(props: ModalPropsType): ReactElement {
-	const { header, onClose, primaryAction, secondaryAction, preventEscaping, children } = props;
+	const { header, onClose, disclosure, primaryAction, secondaryAction, preventEscaping, children } =
+		props;
+	const dialog = useDialogState({ visible: !disclosure });
 	const ref = useRef(null);
```

Comment:
We shouldn't need this, that should be part of Reakit's job

Reply:
Are you're talking about the focus ? It should but it's currently not working https://github.com/ariakit/ariakit/issues/917